corehq/ex-submodules/casexml/apps/case/models.py

```diff
@@ -604,7 +606,19 @@ def apply_updates(self, update_action):
         Applies updates to a case
         """
         for k, v in update_action.updated_known_properties.items():
-            setattr(self, k, v)
+            try:
+                setattr(self, k, v)
+            except BadValueError as e:
+                soft_assert(notify_admins=True)(False, u'Error applying {}: {}'.format(
+                    unicode(update_action), e
+                ))
+                # todo: awkward hard-coded handling for legacy forms - can remove if these
+                # assertions go away: http://manage.dimagi.com/default.asp?182759
+                if k == 'opened_on' and isinstance(v, basestring):
+                    # this will reraise the exception if this wasn't a string date
+                    setattr(self, k, iso_string_to_datetime(v))
```

Comment:
Interesting. So from `k == 'opened_on'` it looks like `date_opened` is converted to `opened_on` in `updated_known_properties`. That makes a lot more sense, and I'm totally on board.